### DISCUSSION TITLE
Copying formulas

### DISCUSSION THREAD

**Discussion by jsimunac01** [ClosedXML/ClosedXML#2124] [Q&A]
Is there a way to enter a formula in the first column of a spreadsheet and have it copy down to the end of the column?  I'm currently selecting the range and using FormuaArrayA1.  But the formula I put in acts like an absolute reference instead of a relative reference, so I get the same result in every cell. 

Thanks for any suggestions.

**Comment by igitur** [comment]
Use FormulaR1C1

**Reply by jsimunac01** [reply]
Using FormulaR1C1 forces it to be an absolute reference.  It shows as $K$2+$L$2 in all fields.  I want K2+L2 on row two, K3+L3 on row 3, etc.  Using FormulaArrayA1 shows as K2+L2 on all rows - without the $.

**Reply by jsimunac01** [reply]
Oopos - didn't show correctly.  Using FormulaR1C1 forces it to be an absolute reference.  It shows K2+L2 (with dollar signs) in all  cells.  Using FormulaArrayA1 shows K2+L2 (without dollar signs) in all cells.  What I want is K2+L2 in row 2, K3+L3 in column 3, etc.

**Reply by jsimunac01** [reply]
I know I can do it by setting a formula and then using a for loop to copy to the last row.  It just seems like there should be a more elegant solution.
